### PR TITLE
Add multi attribute login user resolving to FIDO Authenticator

### PR DESCRIPTION
**Pull request by sandushi**
### Proposed changes in this pull request

- When multi attribute login is enabled, the user needs to be identified using the relevant attributes.
- We need to resolve the user using the multi attribute user resolving logic.
- Check whether the user is already resolved from previous IDF step.  
   - If yes --> don't resolve the user.
   
### This should be merge after
https://github.com/wso2/carbon-identity-framework/pull/4860

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/util/FIDOAuthenticatorConstants.java [hunk 1]
```diff
@@ -44,6 +44,8 @@ private FIDOAuthenticatorConstants() {
 
     public static final String WEBAUTHN_ENABLED = "FIDO.WebAuthn.Enable";
 
+    public static final String IS_USER_RESOLVED = "isUserResolved";
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/util/FIDOAuthenticatorConstants.java:47]: check the possibility of getting this constant from a common constants class for all other authenticators

File: components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java [hunk 2]
```diff
@@ -196,19 +204,65 @@ private AuthenticateRequestData initiateFidoAuthenticationRequest(AuthenticatedU
...
         WebAuthnService webAuthnService = new WebAuthnService();
 
+        if (isPreviousIdPAuthenticationFlowHandler(context)) {
+            boolean isUserResolved = getIsUserResolved(context);
+            if (!isUserResolved) {
+                String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(user.getUserName());
+                String tenantDomain = MultitenantUtils.getTenantDomain(user.getUserName());
+                ResolvedUserResult resolvedUserResult = FrameworkUtils.
+                        processMultiAttributeLoginIdentification(tenantAwareUsername, tenantDomain);
+                if (resolvedUserResult != null && ResolvedUserResult.UserResolvedStatus.SUCCESS.
```
Comment (sandushi) [inline components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java:219]: Check that the user is null

File: components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java [hunk 3]
```diff
@@ -196,19 +204,65 @@ private AuthenticateRequestData initiateFidoAuthenticationRequest(AuthenticatedU
...
+    /**
+     * This method checks if all the authentication steps up to now have been performed by authenticators that
+     * implements AuthenticationFlowHandler interface. If so, it returns true.
+     * AuthenticationFlowHandlers may not perform actual authentication though the authenticated user is set in the
+     * context. Hence, this method can be used to determine if the user has been authenticated by a previous step.
+     *
+     * @param context   AuthenticationContext.
+     * @return True if all the authentication steps up to now have been performed by AuthenticationFlowHandlers.
+     */
+    private boolean isPreviousIdPAuthenticationFlowHandler(AuthenticationContext context) {
```
Comment (sandushi) [inline components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java:245]: move this method to public place and use

File: components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java [hunk 4]
```diff
@@ -196,19 +204,65 @@ private AuthenticateRequestData initiateFidoAuthenticationRequest(AuthenticatedU
...
-    private String initiateFido2AuthenticationRequest(AuthenticatedUser user, String appID)
-            throws AuthenticationFailedException {
+    private String initiateFido2AuthenticationRequest(AuthenticatedUser user, String appID, AuthenticationContext
+            context) throws AuthenticationFailedException {
 
         WebAuthnService webAuthnService = new WebAuthnService();
 
+        if (isPreviousIdPAuthenticationFlowHandler(context)) {
+            boolean isUserResolved = getIsUserResolved(context);
+            if (!isUserResolved) {
```
Comment (sandushi) [inline components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java:214]: Check the user is null can affect to the resolvings

File: components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java [hunk 5]
```diff
@@ -196,19 +204,65 @@ private AuthenticateRequestData initiateFidoAuthenticationRequest(AuthenticatedU
...
+        Map<String, AuthenticatedIdPData> currentAuthenticatedIdPs = context.getCurrentAuthenticatedIdPs();
+        return currentAuthenticatedIdPs != null && !currentAuthenticatedIdPs.isEmpty() &&
+                currentAuthenticatedIdPs.values().stream().filter(Objects::nonNull)
+                        .map(AuthenticatedIdPData::getAuthenticators).filter(Objects::nonNull)
+                        .flatMap(List::stream)
+                        .allMatch(authenticator ->
+                                authenticator.getApplicationAuthenticator() instanceof AuthenticationFlowHandler);
+    }
+
+    private boolean getIsUserResolved(AuthenticationContext context) {
```
Comment (sandushi) [inline components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java:256]: move to a public method

**Comment by sandushi** [comment]
Closed due to duplicate of this https://github.com/wso2-extensions/identity-local-auth-fido/pull/107

**Pull request closed without merging**